src/Evolution/DiscontinuousGalerkin/InboxTags.hpp

```diff
@@ -26,36 +26,42 @@ namespace evolution::dg::Tags {
- * remove boundary data and expect new data to be sent from a neighbor. The
- * TimeStepId also coincides with the next time step of the neighboring
- * element.
+ * face (not the mortar mesh!), the second is the variables at the ghost zone
+ * cells for finite difference/volume reconstruction, and the third is the data
+ * on the mortar needed for computing the boundary corrections (e.g. fluxes,
+ * characteristic speeds, conserved variables). he fourth element of the tuple
+ * is the TimeStepId beyond which the boundary terms are no longer valid, when
+ * using local time stepping.
+ *
+ * The TimeStepId is the neighboring element's next time step. When using local
+ * time stepping, the neighbor's boundary data is valid up until this time,
+ * which may include multiple local time steps. By receiving and storing the
+ * neighbor time step, the local element knows whether or not it should remove
+ * boundary data and expect new data to be sent from the neighbor.
  *
  * The ghost cell data (second element of the tuple) will be valid whenever
- * DG-subcell is being used. Elements using DG and not FD/FV always send both
- * the ghost cells and boundary correction data togther. Elements using FD/FV
- * send the ghost cells first followed by the boundary correction data once the
- * element has received all neighbor ghost cell data.
+ * DG-subcell scheme is being used. Whenever a DG-subcell scheme is being used,
+ * elements using DG and not FD/FV always send both the ghost cells and boundary
+ * correction data together. Elements using FD/FV send the ghost cells first
+ * followed by the boundary correction data once the element has received all
+ * neighbor ghost cell data. Note that the second send/receive only modifies the
+ * flux and the TimeStepId used for the flux validity range.
  *
- * When only DG (not DG-subcell) is used the ghost cell data will never be
- * valid.
+ * When only a DG scheme (not a DG-subcell scheme) is used the ghost cell data
+ * will never be valid.
  *
  * In the DG-subcell scheme this tag is used both for communicating the ghost
  * cell data needed for the FD/FV reconstruction step and the data needed for
- * the boundary corrections. An element using DG will send both the ghost cells
- * and the boundary correction terms simultaneously, while an element using
- * FD/FV will first send the ghost cell data. The element receiving the ghost
- * cells will store the data in this inbox tag and be able to do reconstruction
- * once all neighboring ghost cells are received. In the mean time, as boundary
- * correction data is received from neighbors it is added to this inbox tag. The
- * inbox tag is not erased until after the boundary correction computation is
- * complete. The reason for this design choice is that it is mostly the number
- * of communications that dominates cost, not the size of the communication.
+ * the boundary corrections. An element using a DG scheme will send both the
```

Comment:
here I think "DG scheme" -> "DG" because we're talking about DG timesteps within the DG-subcell scheme? I don't know if this "convention" is any good, but this change would at least make it a bit more consistent.